patches/tModLoader/Terraria/Program.TML.cs

```diff
@@ -120,6 +120,44 @@ private static void SetSavePath()
 			Logging.tML.Info($"Controlled Folder Access feature detected. If game fails to launch make sure to add \"{Environment.ProcessPath}\" to the \"Allow an app through Controlled folder access\" menu found in the \"Ransomware protection\" menu."); // Before language is loaded, no need to localize
 	}
 
+	private static void StartupSequenceTml(bool isServer)
+	{
+		try {
+			ControlledFolderAccessSupport.CheckFileSystemAccess();
+			Logging.Init(isServer ? Logging.LogFile.Server : Logging.LogFile.Client);
+
+			if (Platform.Current.Type == PlatformType.Windows && System.Runtime.InteropServices.RuntimeInformation.ProcessArchitecture != System.Runtime.InteropServices.Architecture.X64)
+				throw new Exception("The current Windows Architecture of your System is CURRENTLY unsupported. Aborting...");
+		}
+		catch (Exception e) {
+			ErrorReporting.FatalExit("Failed to init logging", e);
+		}
+
+		Logging.LogStartup(isServer); // Should run as early as is possible. Want as complete a log file as possible
+
+		try {
+			SetSavePath(); 
+		}
+		catch (Exception e) {
+			ErrorReporting.FatalExit("Failed to establish a save location", e);
+		}
+				
+		if (ModLoader.Core.ModCompile.DeveloperMode) // Needs to run after SetSavePath
```

Comment:
Not just needs to run after `SetSavePath`, needs to have static ctor called after `SetSavePath`, so might need to check that via breakpoint

Reply:
Tested and validated that the ModComiple static ctor was only invoked by this in LogStartup. 
This ordering works ok.